### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by BasketOfEggs**
Tokens

### CODE REVIEW

File: backend/src/main/java/com/senior/project/backend/security/AuthHandler.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+     */
+    public Mono<ServerResponse> signIn(ServerRequest req) {
+        return req.bodyToMono(LoginRequest.class)
+            .flatMap(request -> {
+                String idToken = request.getIdToken();
+                TokenType type = request.getType();
+                try {
+                    TokenVerifier verifier = this.tokenVerifierGetter.getTokenVerifier(type);
+                    String oid = verifier.verifiyIDToken(idToken);
+                    // verifier.verifiyAccessToken(accessToken);
```
Comment (jts7382) [inline backend/src/main/java/com/senior/project/backend/security/AuthHandler.java:58]: why is this commented out?

File: backend/build.gradle.kts [hunk 2]
```diff
@@ -23,6 +23,9 @@ dependencies {
     compileOnly("org.projectlombok:lombok:1.18.20")
     annotationProcessor("org.projectlombok:lombok:1.18.20")
     implementation("org.projectlombok:lombok:1.18.28")
+    implementation("org.bitbucket.b_c:jose4j:0.6.0")
```
Comment (jts7382) [inline backend/build.gradle.kts:26]: why do we need bitbucket?

**Review by rcb2957 — APPROVED** [review]
Appears that authentication features have been successfully integrated

File: frontend/crd/src/app/milestones-page/milestones/milestone.service.ts [hunk 3]
```diff
@@ -15,7 +16,7 @@ export class MilestoneService {
 
   // Get all data items
   getMilestones(): Observable<Milestone[]> {
-    return this.http.get<Milestone[]>('http://localhost:8080/api/milestones')
+    return this.http.get<Milestone[]>(constructBackendRequest(Endpoints.MILESTONES))
```
Comment (jts7382) [inline frontend/crd/src/app/milestones-page/milestones/milestone.service.ts:19]: Thanks. Was thinking we needed this

File: frontend/crd/src/app/security/user.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export interface TempUser {
+    readonly name: string;
+    readonly email: string;
+    readonly role: string;
```
Comment (jts7382) [inline frontend/crd/src/app/security/user.ts:4]: We should have an enum for roles (user, faculty, admin)

**Review by jts7382 — APPROVED** [review]
Looks good, but didn't get to test it yet

File: backend/src/main/java/com/senior/project/backend/security/AuthHandler.java (hunk 1 above)
Comment (BasketOfEggs) [reply backend/src/main/java/com/senior/project/backend/security/AuthHandler.java:58]: I need to delete that line, that method no longer exists

File: backend/build.gradle.kts (hunk 2 above)
Comment (BasketOfEggs) [reply backend/build.gradle.kts:26]: Thats the source for the library, its not bitbucket but jose4j, which is a library for handling JWTs

File: frontend/crd/src/app/security/user.ts (hunk 4 above)
Comment (BasketOfEggs) [reply frontend/crd/src/app/security/user.ts:4]: At some point I agree, this is just a temporary class for testing purposes that will be replaced by the actual user object

**Comment by bml2238** [comment]
any reason why this hasn't been merged?

**Pull request merged**